packages/apollo/src/commands/service/__tests__/list.test.ts

```diff
@@ -0,0 +1,229 @@
+import ServiceList from "../list";
+import chalk from "chalk";
+import { stdout } from "stdout-stderr";
+import nock = require("nock");
+
+const stagingAPI = "https://engine-staging-graphql.apollographql.com:443";
+
+/**
+ * Default API key. This is not an actual API key but a randomly generated string.
+ *
+ * If you need to use the `nock` recorder, then this will not work because we won't be able to access engine
+ * with a fake API key.
+ */
+const fakeApiKey = "service:engine:9YC5AooMa2yO11eFlZat11";
+
+/**
+ * Engine API key we're using.
+ *
+ * This is hard-coded to `fakeApiKey` because this is out day-to-day usage should be. If we're going to be
+ * updating the mocked data; we'll need to use a real API key (see [README#Regenerating Mocked Network
+ * Data](https://github.com/apollographql/apollo-tooling#regenerating-mocked-network-data)); which will be
+ * placed here.
+ */
+const apiKey = fakeApiKey;
+
+/**
+ * An array that we'll spread into all CLI commands to pass the engine api key.
+ */
+const cliKeyParameter = [`--key=${apiKey}`];
+
+/**
+ * The original `console.log` being mocked.
+ *
+ * We save it so we can restore it after a test.
+ */
+let mockedConsoleLogOriginal: Console["log"] | null = null;
+
+/**
+ * Array of intercepted console values.
+ */
+let mockedConsoleLogValues: string[] | null = null;
+
+/**
+ * Mock and capture `console.log` and `stdout.write`s. Return them in that order as a single string.
+ *
+ * This will emulate what the output of running the CLI would look like.
+ *
+ * Call `uncaptureApplicationOutput` to reverse the effects of this function.
+ */
+function captureApplicationOutput() {
```

Comment:
this and the following function are copied from `check.test.ts` I am not sure how to share them with a shared `console.log`

Reply:
hmmm... I think this would actually hugely benefit from passing in a custom logger to the entirety of the `apollo` CLI. Probably a feature we should log somewhere, if it's not already. That way you can just instantiate the tests with a `writeToMap` logger that stores the log output in an array under the test name or something.

Reply:
@zionts, agree on both fronts - definitely a feature we should consider adding. In the meantime, @mayakoneval can you add comments in both places that this code is duplicated by necessity? Enough to provide some context for future implementors, as they may have to do the same thing.